File: app/src/main/java/com/github/sdpcoachme/firebase/auth/GoogleAuthComponent.java
package com.github.sdpcoachme.firebase.auth;

import dagger.Component;

/**
 * Component that provides the dependencies for the GoogleAuthenticator
 */
@Component(modules = {GoogleAuthModule.class})
public interface GoogleAuthComponent {

    /**
     * Injects the dependencies into the given activity
     *
     * @param firebaseUIActivity the activity to inject into
     */
    void inject(FirebaseUIActivity firebaseUIActivity);
}

File: app/src/main/java/com/github/sdpcoachme/firebase/auth/GoogleAuthComponent.kt
package com.github.sdpcoachme.firebase.auth

import dagger.Component

/**
 * Component that provides the dependencies for the GoogleAuthenticator
 */
@Component(modules = [GoogleAuthModule::class])
interface GoogleAuthComponent {
    /**
     * Injects the dependencies into the given activity
     *
     * @param firebaseUIActivity the activity to inject into
     */
    fun inject(firebaseUIActivity: FirebaseAuthActivity?)
}